fury/utils.py

```diff
@@ -1547,6 +1548,55 @@ def color_check(pts_len, colors=None):
     return color_array, global_opacity
 
 
+def check_color_range(color_array):
+    """
+    Check if all values in a color array are within the range [0, 1].
+    If there is out of bounds values, the function sends a message, 
+    replaces the out-of-bounds colors with the red color [1, 0, 0].
+
+    Args:
+        color_array (numpy.ndarray): An array of shape (N, 3) or (N, 4).
+
+    Returns:
+        The orignal array if all values are within the range [0,1].
+        Otherwise, the updated numpy.ndarray of color_array
+    """
+    # Keep the option for color = None for some actors
+    if color_array is None:
+        return color_array
+
+    # Convert tuple or list to ndarray (N, 3)
+    if isinstance(color_array, list):
+        color_array = np.asarray(color_array)
+
+    is_tuple = False
+    if isinstance(color_array, tuple):
+        is_tuple = True
+        color_array = np.asarray(color_array)
```

Comment:
`isinstance(color_aray, (list, tuple))` instead

Reply:
It was `isinstance(color_aray, (list, tuple))`, but when I went through all the test, I found it is a requirement for input 1d list keeps as a 1d list and input tuple keeps a tuple. Therefore you can find in the end of the function, there is one line to convert np.ndarray back to tuple if it is the case.

Reply:
It is required for in the color_check() function in actor.py
```
def color_check(pts_len, colors=None):
        """
        Returns a VTK scalar array containing colors information for each one of
        the points according to the policy defined by the parameter colors.

        Parameters
        ----------
        pts_len : int
            length of points ndarray
        colors : None or tuple (3D or 4D) or array/ndarray (N, 3 or 4)
            If None a predefined color is used for each point.
            **If a tuple of color is used. Then all points will have the same color.
            If an array (N, 3 or 4) is given, where N is equal to the number of
            points. Then every point is colored with a different RGB(A) color.**

        Returns
        -------
        color_array : vtkDataArray
            vtk scalar array with name 'colors'.
        global_opacity : float
            returns 1 if the colors array doesn't contain opacity otherwise -1.
            If colors array has 4 dimensions, it checks values of the fourth
            dimension. If the value is the same, then assign it to global_opacity.

        """
```

Reply:
Have applied `isinstance(color_aray, (list, tuple)`